src/deploy/functions/tasks.ts

```diff
@@ -110,122 +124,92 @@ export function updateFunctionTask(
-      params.timer.endTimer(fnName);
-      helper.printSuccess(fnName, "delete");
-      return operationResult;
-    } catch (err) {
-      params.errorHandler.record("error", fnName, "delete", err.message || "");
-    }
+    const deleteRes = await cloudfunctions.deleteFunction({
+      functionName: fnName,
+    });
+    const pollerOptions: OperationPollerOptions = Object.assign(
+      {
+        pollerName: `delete-${fnName}`,
+        operationResourceName: deleteRes.name,
+      },
+      defaultPollerOptions
+    );
+    return await pollOperation<void>(pollerOptions);
   };
+  task.functionName = fnName;
+  task.operationType = "delete";
+  return task;
 }
 
-export function upsertScheduleTask(
-  params: TaskParams,
-  fn: CloudFunctionTrigger,
-  appEngineLocation: string
-): () => Promise<any> {
-  return async () => {
-    const job = helper.toJob(fn, appEngineLocation, params.projectId);
+export function functionsDeploymentHandler(
+  timer: DeploymentTimer,
+  errorHandler: ErrorHandler
+): (task: DeploymentTask) => Promise<any | undefined> {
+  return async (task: DeploymentTask) => {
+    let result;
     try {
-      await cloudscheduler.createOrReplaceJob(job);
-      helper.printSuccess(fn.name, "upsert schedule");
+      timer.startTimer(task.functionName, task.operationType);
+      result = await task();
+      helper.printSuccess(task.functionName, task.operationType);
     } catch (err) {
-      params.errorHandler.record("error", fn.name, "upsert schedule", err.message || "");
-    }
-  };
-}
-
-export function deleteScheduleTask(
-  params: TaskParams,
-  fnName: string,
-  appEngineLocation: string
-): () => Promise<void> {
-  return async () => {
-    const jobName = helper.getScheduleName(fnName, appEngineLocation);
-    const topicName = helper.getTopicName(fnName);
-    try {
-      await cloudscheduler.deleteJob(jobName);
-    } catch (err) {
-      // If the job has already been deleted, don't throw an error.
-      if (err.status !== 404) {
-        params.errorHandler.record("error", fnName, "delete schedule", err.message || "");
-        return;
+      if (err.original?.context?.response?.statusCode === 429) {
+        // Throw quota errors so that throttler retries them.
+        throw err;
       }
-      logger.debug(`Scheduler job ${jobName} not found.`);
-    }
-    try {
-      await deleteTopic(topicName);
-      helper.printSuccess(fnName, "delete schedule");
-    } catch (err) {
-      // If the topic has already been deleted, don't throw an error.
-      if (err.status !== 404) {
-        params.errorHandler.record("error", fnName, "delete schedule", err.message || "");
-        return;
-      }
-      logger.debug(`Scheduler topic ${topicName} not found.`);
+      errorHandler.record(
+        "error",
+        task.functionName,
+        task.operationType,
+        err.original?.message || ""
+      );
     }
+    timer.endTimer(task.functionName);
```

Comment:
I don't think it matters because the only time we don't hit this line is if we throw a quota error, and the Queue will retry on errors, so endTimer will eventually be called.